src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -282,6 +309,33 @@ public sealed class AddPathCommandExtension : RunnerService, IActionCommandExten
 
         public void ProcessCommand(IExecutionContext context, string line, ActionCommand command, ContainerInfo container)
         {
+            var serverUrl = context.GetGitHubContext("server_url");
+            var isNotGithub = false;
+            if(!String.IsNullOrWhiteSpace(serverUrl) && serverUrl != "https://github.com")
+            {
+                isNotGithub = true;
+            }
+            
+            var allowUnsecureCommands = false;
+            bool.TryParse(Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowUnsupportedCommands), out allowUnsecureCommands);
+
+            // TODO: Eventually remove isNotGithub and apply this to dotcom customers as well
+            if (isNotGithub && !allowUnsecureCommands)
+            {
+                throw new Exception(String.Format(Constants.Runner.UnsupportedCommandMessageDisabled, this.Command));
+            }
+            else if(!allowUnsecureCommands)
```

Comment:
kind of weird that `allow` is false, and then we warn but allow anyway

wondering whether the default value of `allow` should be based on whether is hosted server. And then if the env var is specified, respect it.